[BUG] Create error from restore resource

### Terraform CLI and Terraform TiDB Cloud Provider Version
```shel
$ terraform version
Terraform v1.6.5
on darwin_arm64
+ provider registry.terraform.io/tidbcloud/tidbcloud v0.2.1
```

### Describe the bug
`tidbcloud_restore` returns the following `Create error` with a high probability. I got the error 9 out of 10.
```shell
│ Error: Create Error
│
│   with tidbcloud_restore.example_restore,
│   on main.tf line 16, in resource "tidbcloud_restore" "example_restore":
│   16: resource "tidbcloud_restore" "example_restore" {
│
│ Unable to call GetRestoreTask, got error: [GET /api/v1beta/projects/{project_id}/restores/{restore_id}][500] getRestoreTaskInternalServerError
│ &{Code:159900104 Details:[] Message:internal error}
╵
```
But actually, all the restore tasks run, and it's successfully completed.

I assume this is because `GetRestoreTask` is called immediately after `CreateRestoreTask`.
https://github.com/tidbcloud/terraform-provider-tidbcloud/blob/83ce73a897c9fbc5eab242a41919754c468163b7/internal/provider/restore_resource.go#L262-L274

Namely, the tidbcloud provider calls `GetRestoreTask` before it is ready to be called. A wait or retry mechanism might be required.

### Steps to Reproduce
<!--
Please list the full steps required to reproduce the issue, for example:
1. `terraform init`
2. `terraform init
-->
1. Prepare the tf file to restore the cluster.
    - See the example at https://docs.pingcap.com/tidbcloud/terraform-use-restore-resource#create-a-restore-task
2. `terraform init`
3. `terraform apply`

Hi, I tried, but I got the plugin crash 3 out of 3. Could you please take a look at the following stack trace? Thanks!

```
$ tf apply plan.out
tidbcloud_restore.example_restore: Creating...
╷
│ Error: Plugin did not respond
│
│ The plugin encountered an error, and failed to respond to the plugin6.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain more
│ details.
╵

Stack trace from the terraform-provider-tidbcloud_v0.2.1 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x104c864a0]

goroutine 50 [running]:
github.com/tidbcloud/terraform-provider-tidbcloud/internal/provider.refreshRestoreResourceData(0x140006004d0, 0x14000202200)
        github.com/tidbcloud/terraform-provider-tidbcloud/internal/provider/restore_resource.go:364 +0x110
github.com/tidbcloud/terraform-provider-tidbcloud/internal/provider.restoreResource.Create({0x0?}, {0x104fbbe40, 0x140005b36b0}, {{{{0x104fbf4e8, 0x14000626cc0}, {0x104ea0cc0, 0x14000564c60}}, {0x104fc1368, 0x140000c2820}}, {{{0x104fbf4e8, ...}, ...}, ...}, ...}, ...)
        github.com/tidbcloud/terraform-provider-tidbcloud/internal/provider/restore_resource.go:275 +0x460
github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(*Server).CreateResource(0x14000211080, {0x104fbbe40, 0x140005b36b0}, 0x14000737330, 0x140007372d0)
        github.com/hashicorp/terraform-plugin-framework@v1.1.1/internal/fwserver/server_createresource.go:97 +0x420
github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(*Server).ApplyResourceChange(0x104fbbe40?, {0x104fbbe40, 0x140005b36b0}, 0x140005852c0, 0x140007374a0)
        github.com/hashicorp/terraform-plugin-framework@v1.1.1/internal/fwserver/server_applyresourcechange.go:54 +0x37c
github.com/hashicorp/terraform-plugin-framework/internal/proto6server.(*Server).ApplyResourceChange(0x14000211080, {0x104fbbe40?, 0x140005b3560?}, 0x14000585270)
        github.com/hashicorp/terraform-plugin-framework@v1.1.1/internal/proto6server/server_applyresourcechange.go:52 +0x1a8
github.com/hashicorp/terraform-plugin-go/tfprotov6/tf6server.(*server).ApplyResourceChange(0x1400047e780, {0x104fbbe40?, 0x140005b2d80?}, 0x140004aefc0)
        github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov6/tf6server/server.go:816 +0x3c0
github.com/hashicorp/terraform-plugin-go/tfprotov6/internal/tfplugin6._Provider_ApplyResourceChange_Handler({0x104f845a0?, 0x1400047e780}, {0x104fbbe40, 0x140005b2d80}, 0x140004aef50, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov6/internal/tfplugin6/tfplugin6_grpc.pb.go:385 +0x174
google.golang.org/grpc.(*Server).processUnaryRPC(0x1400065a1e0, {0x104fbfb28, 0x14000093040}, 0x140005a8c60, 0x140003edb60, 0x1055313c0, 0x0)
        google.golang.org/grpc@v1.51.0/server.go:1340 +0xb90
google.golang.org/grpc.(*Server).handleStream(0x1400065a1e0, {0x104fbfb28, 0x14000093040}, 0x140005a8c60, 0x0)
        google.golang.org/grpc@v1.51.0/server.go:1713 +0x840
google.golang.org/grpc.(*Server).serveStreams.func1.2()
        google.golang.org/grpc@v1.51.0/server.go:965 +0x88
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.51.0/server.go:963 +0x298

Error: The terraform-provider-tidbcloud_v0.2.1 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

My Terraform version:
```
x terraform version
Terraform v1.6.5
on darwin_arm64
+ provider registry.terraform.io/tidbcloud/tidbcloud v0.2.1
```

--- a/internal/provider/restore_resource.go
+++ b/internal/provider/restore_resource.go
@@ -360,10 +360,12 @@ func refreshRestoreResourceData(resp *restoreApi.GetRestoreTaskOKBody, data *res
 	data.RestoreId = types.StringValue(resp.ID)
 	data.CreateTimestamp = types.StringValue(resp.CreateTimestamp.String())
 	data.Status = types.StringValue(resp.Status)
-	data.Cluster = &cluster{
-		Id:     resp.Cluster.ID,
-		Name:   resp.Cluster.Name,
-		Status: resp.Cluster.Status,
+	if resp.Cluster != nil {
+		data.Cluster = &cluster{
+			Id:     resp.Cluster.ID,
+			Name:   resp.Cluster.Name,
+			Status: resp.Cluster.Status,
+		}
 	}
 	data.ErrorMessage = types.StringValue(resp.ErrorMessage)
 }

